pkg/api/validation/validation.go

```diff
@@ -1398,3 +1400,90 @@ func ValidateEndpointsUpdate(oldEndpoints, newEndpoints *api.Endpoints) errs.Val
 	allErrs = append(allErrs, validateEndpointSubsets(newEndpoints.Subsets).Prefix("subsets")...)
 	return allErrs
 }
+
+// ValidateSecurityContext tests to make sure that a security context is set and that the settings
+// in the security context do not conflict with settings on the container.
+func ValidateSecurityContext(container *api.Container) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	mergeSecurityContextAndContainer(container)
+	sc := container.SecurityContext
+	if container.Privileged != *sc.Privileged {
+		allErrs = append(allErrs, errs.NewFieldInvalid("privileged", *sc.Privileged, "privileged conflicts with container.privileged"))
+	}
+	if !reflect.DeepEqual(sc.Capabilities.Add, container.Capabilities.Add) {
+		allErrs = append(allErrs, errs.NewFieldInvalid("capabilities.add", sc.Capabilities.Add, "capabilities.add conflicts with container.capabilities.add"))
+	}
+	if !reflect.DeepEqual(sc.Capabilities.Drop, container.Capabilities.Drop) {
+		allErrs = append(allErrs, errs.NewFieldInvalid("capabilities.drop", sc.Capabilities.Drop, "capabilities.drop conflicts with container.capabilities.drop"))
+	}
+	if sc.SELinuxOptions != nil {
+		allErrs = append(allErrs, validateSELinuxOptions(sc.SELinuxOptions).Prefix("seLinuxOptions")...)
+	}
+	if sc.RunAsUser != nil {
+		if *sc.RunAsUser < 0 {
+			allErrs = append(allErrs, errs.NewFieldInvalid("runAsUser", *sc.RunAsUser, "runAsUser cannot be negative"))
+		}
+	}
+	return allErrs
+}
+
+// validateSELinux ensures that the disable and label options are not mixed
+func validateSELinuxOptions(seLinux *api.SELinuxOptions) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	if seLinux.Disabled {
+		if len(seLinux.Level) > 0 || len(seLinux.Role) > 0 || len(seLinux.Type) > 0 || len(seLinux.User) > 0 {
+			allErrs = append(allErrs, errs.NewFieldInvalid("disabled", seLinux.Disabled, "cannot set disabled and labels"))
+		}
+	}
+	return allErrs
+}
+
+// mergeSecurityContextAndContainer performs a merge of the settings that exist on both the container
+// and the securitycontext.  This includes the Privileged field and Capabilities field
+// merge rules are:
+// 1.  if the container has no security context then create one and use all settings from the container
```

Comment:
Should we just not create a context at all if privileged=false and capabilities to add and drop are empty lists?